tests/conftest.cpp

```diff
@@ -0,0 +1,376 @@
+
+      // set fullPythonPath and fullPythonBinPath
+      fullPythonPath = fullVenvPath / binPath / pythonPath;
+      fullPythonBinPath = fullVenvPath / binPath;
+
+      std::cout << "DocsPythonVenvTest:: fullPythonPath: " << fullPythonPath << std::endl;
+      std::cout << "DocsPythonVenvTest:: fullPythonBinPath: " << fullPythonBinPath << std::endl;
+
+      std::filesystem::path requirements_path = std::filesystem::path(TESTS_DOCS_DIR) / "requirements.txt";
+
+      // array of commands to run
+      std::string CommandArray[] = {
+        "\"" + fullPythonPath.string() + "\" -m pip install -r " + requirements_path.string(),
+      };
+
+      for (const std::string& _command : CommandArray) {
+        std::cout << "DocsPythonVenvTest:: running command: " << _command << std::endl;
+        exit_code = BaseTest::exec(_command.c_str());
+        if (exit_code != 0) {
+          FAIL() << "Command failed: " << command << " with exit code: " << exit_code;
+        }
+      }
+
+      // Save the original PATH
+      originalEnvPath = std::getenv("PATH") ? std::getenv("PATH") : "";
+      std::cout << "DocsPythonVenvTest:: originalEnvPath: " << originalEnvPath << std::endl;
+
+      // Set the temporary PATH
+      std::string tempPath;
+      std::string envPathSep;
+
+  #ifdef _WIN32
+      envPathSep = ";";
+  #else
+      envPathSep = ":";
+  #endif
+      tempPath = fullPythonBinPath.string() + envPathSep + originalEnvPath;
+      std::cout << "DocsPythonVenvTest:: tempPath: " << tempPath << std::endl;
+      setEnv("PATH", tempPath);
+
+      std::cout << "DocsPythonVenvTest:: finished Fixture SetUp" << std::endl;
+    }
+
+    void
+    TearDown() override{
+      std::cout << "DocsPythonVenvTest:: starting Fixture TearDown" << std::endl;
+
+      // Restore the original PATH
+      if (!originalEnvPath.empty()) {
+        std::cout << "DocsPythonVenvTest:: restoring originalEnvPath: " << originalEnvPath << std::endl;
+        setEnv("PATH", originalEnvPath);
+      }
+
+      std::cout << "DocsPythonVenvTest:: finished Fixture TearDown" << std::endl;
+    }
+#else
+    void
+    SetUp() override{
+      std::cout << "DocsPythonVenvTest:: starting Fixture SetUp" << std::endl;
+      std::cout << "DocsPythonVenvTest:: skipping test because TESTS_ENABLE_VENV_TESTS is not set" << std::endl;
+      std::cout << "DocsPythonVenvTest:: finished Fixture SetUp" << std::endl;
+    }
+
+    void
+    TearDown() override{
+      std::cout << "DocsPythonVenvTest:: starting Fixture TearDown" << std::endl;
+      std::cout << "DocsPythonVenvTest:: skipping test because TESTS_ENABLE_VENV_TESTS is not set" << std::endl;
+      std::cout << "DocsPythonVenvTest:: finished Fixture TearDown" << std::endl;
+    }
+#endif
+
+    // functions and variables
+    std::filesystem::path fullPythonPath;
+    std::filesystem::path fullPythonBinPath;
+    std::string originalEnvPath;
+};
+
+class DocsPythonVenvTest : public virtual BaseTest, public DocsPythonVenvBase {
```

Comment:
I'm not really a fan of these secondary classes, but I don't know how to make it work without them.